### PR TITLE
Switch to build.sbt style since Build.scala is deprecated

### PR DESCRIPTION
**Pull request by LogicalTime**
I made a version that uses the build.sbt style since Build.scala is deprecated. It also uses the Def.task{} style of defining a task which some coders might like more.

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @LogicalTime,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: project/Build.scala [hunk 1]
```diff
@@ -1,52 +0,0 @@
...
-import Tests._
-
-/** 
- *  This is a slightly more advanced sbt setup using two projects.
- *  The first one, "codegen" a customized version of Slick's
- *  code-generator. The second one "main" depends on "codegen", which means
- *  it is compiled after "codegen". "main" uses the customized
- *  code-generator from project "codegen" as a sourceGenerator, which is run
- *  to generate Slick code, before the code in project "main" is compiled.
- */
```
Comment (cvogt) [inline project/Build.scala]: This comment should stay

File: project/Build.scala [hunk 2]
```diff
@@ -1,52 +0,0 @@
...
-object stagedBuild extends Build {
-  /** main project containing main source code depending on slick and codegen project */
```
Comment (cvogt) [inline project/Build.scala]: this one probably too

File: build.sbt [hunk 3]
```diff
@@ -0,0 +1,40 @@
+
+fork in run := true
```
Comment (cvogt) [inline build.sbt]: why this? It breaks `-jvm-debug`, so unless there is a reason, I wouldn't put it into the example.

File: build.sbt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+lazy val root = (project in file("."))
+    .settings(sharedSettings)
+    .settings(slick := slickCodeGenTask.value) // register manual sbt command)
+    .settings(sourceGenerators in Compile += slickCodeGenTask.taskValue) // register automatic code generation on every compile, remove for only manual use)
+    .dependsOn(codegen)
+
+
+/** codegen project containing the customized code generator */
+lazy val codegen = project
```
Comment (cvogt) [inline build.sbt]: how does sbt know the folder name?

Comment (LogicalTime) [inline build.sbt]: for the codegen project? 

> A project is defined by declaring a lazy val of type Project. For example, :
> 
> lazy val util = project
> 
> lazy val core = project
> The name of the val is used as the project’s ID and base directory name. The ID is used to refer to the project at the command line. The base directory may be changed from the default using the in method. For example, the following is a more explicit way to write the previous example:
> 
> lazy val util = project.in(file("util"))
> 
> lazy val core = project in file("core")
> 

http://www.scala-sbt.org/0.13/docs/Multi-Project.html

File: build.sbt (hunk 3 above)
Comment (LogicalTime) [inline build.sbt]: sure, I'll get rid of it. I didn't mean to add any settings not in the original Build.scala file.

File: project/Build.scala (hunk 1 above)
Comment (LogicalTime) [inline project/Build.scala]: Whoops, will fix.

File: project/Build.scala (hunk 2 above)
Comment (LogicalTime) [inline project/Build.scala]: will fix

**Comment by LogicalTime** [comment]
How does it look now?

**Comment by LogicalTime** [comment]
Anything else I should do? This is one of my first pull requests.

File: build.sbt (hunk 4 above)
Comment (cvogt) [inline build.sbt]: I see. `.sbt` is even less `.scala` than I thought ;).

**Comment by cvogt** [comment]
there was still a compile error with the :Unit change. I fixed it in #3 and merged. Thx for being part of this community and contributing :)!

**Pull request closed without merging**